Sources/firebase/include/FirebaseCore.hh

```diff
@@ -1,7 +1,34 @@
-
 #ifndef firebase_include_FirebaseCore_hh
 #define firebase_include_FirebaseCore_hh
 
 #include <firebase/util.h>
 
+#include <stdio.h>
+
+namespace swift_firebase::swift_cxx_shims::firebase {
+
+typedef void (*FutureCompletionType)(void*);
+
+// This class exists to provide protocol conformance to FutureProtocol.  It
+// also provides a method to invoke `OnCompletion` in a way that works from
+// Swift. We can ignore the `FutureBase` param as the Swift caller can just
+// retain the Future as part of its closure.
+template <class R> class ConformingFuture: public ::firebase::Future<R> {
```

Comment:
I don't understand why this type is called a `ConformingFuture`. Can we not simply name this `Future`?

Reply:
That felt confusing to me. There is already something called `Future`. Why do you want to overload the name?

Reply:
Because the type itself is an extension to the `Future` type. The name `ConformingFuture` doesn't really say anything. I think that at the very least, this should be renamed to something that reflects what this type is. I suppose that if that was clarified not overloading the `Future` type from Firebase would be reasonable.

Reply:
That was the purpose of the name `ConformingFuture`. It exists mainly to provide conformance to `FutureProtocol`. Can also use a name like `FutureWithFutureProtocolConformance` but that sounds like a mouthful.

Reply:
I went with just calling it `Future` instead of trying to be more descriptive.